mlrun/api/api/endpoints/functions.py

```diff
@@ -500,40 +502,64 @@ def _handle_job_deploy_status(
             content=out,
             media_type="text/plain",
             headers={
-                "x-mlrun-function-status": state,
-                "function_status": state,
+                "x-mlrun-function-status": function_state,
+                "function_status": function_state,
                 "function_image": image,
                 "builder_pod": pod,
             },
         )
 
-    # TODO: change state to pod_status
-    state = mlrun.api.utils.singletons.k8s.get_k8s_helper(silent=False).get_pod_status(
-        pod
+    build_pod_state = mlrun.api.utils.singletons.k8s.get_k8s_helper(
+        silent=False
+    ).get_pod_status(pod)
+    logger.debug(
+        "Resolved pod status",
+        function_name=name,
+        pod_status=build_pod_state,
+        pod_name=pod,
     )
-    logger.info("Resolved pod status", pod_status=state, pod_name=pod)
 
-    if state == "succeeded":
-        logger.info("Build completed successfully")
-        state = mlrun.common.schemas.FunctionState.ready
-    if state in ["failed", "error"]:
-        logger.error("Build failed", pod_name=pod, pod_status=state)
-        state = mlrun.common.schemas.FunctionState.error
+    normalized_build_pod_state = (
+        mlrun.common.schemas.FunctionState.get_function_state_from_pod_state(
+            build_pod_state
+        )
+    )
+    if normalized_build_pod_state == mlrun.common.schemas.FunctionState.ready:
```

Comment:
one is function state and other is its pod state?
can one be failed and other ready? 👁️

Reply:
I'll change it to `normalized_pod_function_state`